TELCODOCS-2045 - Telco core RDS 4.18 docs (#88622)
openshift/openshift-docs: modules/telco-core-application-workloads.adoc

When creating pod configurations that require exclusive CPUs, be aware of the potential implications of hyper-threaded systems.
Pods should request multiples of 2 CPUs when the entire core (2 hyper-threads) must be allocated to the pod.

Pods running network functions that do not require high throughput or low latency networking should be scheduled with best-effort or burstable QoS pods and do not require dedicated or isolated CPU cores.

Engineering considerations::
+
--
Use the following information to plan telco core workloads and cluster resources.

* CNF applications should conform to the latest version of https://redhat-best-practices-for-k8s.github.io/guide/[Red Hat Best Practices for Kubernetes].
* Use a mix of best-effort and burstable QoS pods as required by your applications.
** Use guaranteed QoS pods with proper configuration of reserved or isolated CPUs in the `PerformanceProfile` CR that configures the node.
** Guaranteed QoS Pods must include annotations for fully isolating CPUs.
** Best effort and burstable pods are not guaranteed exclusive use of a CPU.

> Workloads may be preempted by other workloads, OS daemons, or kernel tasks.

Reviewer: Per ISG, use _may_ only to indicate permission.  
```suggestion
Workloads can be preempted by other workloads, OS daemons, or kernel tasks.
```

> * Use exec probes sparingly and only when no other suitable options are available.

Reviewer: The telco-core-ref-application-workloads module uses similar wording in the singular (Exec probes should be avoided unless there is no viable alternative.)
```suggestion
* Use exec probes sparingly and only when no other suitable option is available.
```

** Do not use exec probes if a CNF uses CPU pinning.
Use other probe implementations, for example, `httpGet` or `tcpSocket`.
** When you need to use exec probes, limit the exec probe frequency and quantity.
The maximum number of exec probes must be kept below 10, and frequency must not be set to less than 10 seconds.

> ** You can use startup probes, since they do not use significant resources at steady-state operation.

Reviewer: Per ISG, [use _since_ only to indicate time](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#s).

```suggestion
** You can use startup probes, because they do not use significant resources at steady-state operation.
```

> The limitation on exec probes applies primarily to liveness and readiness probes.

Reviewer: Maybe be specific so that people don't think that all limitations on exec probes apply to liveness and readiness probes here?

```suggestion
This limitation on exec probes applies primarily to liveness and readiness probes.
```

When creating pod configurations that require exclusive CPUs, be aware of the potential implications of hyper-threaded systems.
Pods should request multiples of 2 CPUs when the entire core (2 hyper-threads) must be allocated to the pod.

Pods running network functions that do not require high throughput or low latency networking should be scheduled with best-effort or burstable QoS pods and do not require dedicated or isolated CPU cores.

Engineering considerations::
+
--
Use the following information to plan telco core workloads and cluster resources.

* CNF applications should conform to the latest version of https://redhat-best-practices-for-k8s.github.io/guide/[Red Hat Best Practices for Kubernetes].
* Use a mix of best-effort and burstable QoS pods as required by your applications.
** Use guaranteed QoS pods with proper configuration of reserved or isolated CPUs in the `PerformanceProfile` CR that configures the node.
** Guaranteed QoS Pods must include annotations for fully isolating CPUs.

> ** Best effort and burstable pods are not guaranteed exclusive use of a CPU.

Reviewer 2: ```suggestion
** Best effort and burstable pods are not guaranteed exclusive CPU use.
```

Workloads can be preempted by other workloads, operating system daemons, or kernel tasks.
* Use exec probes sparingly and only when no other suitable option is available.
** Do not use exec probes if a CNF uses CPU pinning.
Use other probe implementations, for example, `httpGet` or `tcpSocket`.
** When you need to use exec probes, limit the exec probe frequency and quantity.

> The maximum number of exec probes must be kept below 10, and frequency must not be set to less than 10 seconds.

Reviewer 2: ```suggestion
The maximum number of exec probes must be kept below 10, and the frequency must not be set to less than 10 seconds.
```
